src/community/misc.c

```diff
@@ -120,12 +131,24 @@ int igraph_community_to_membership(const igraph_matrix_t *merges,
         igraph_vector_null(csize);
     }
 
+    IGRAPH_VECTOR_INIT_FINALLY(&already_merged, steps + no_of_nodes);
```

Comment:
Since you only ever going to store `0` or `1` in this vector, an `igraph_vector_bool_t` would suffice.

Reply:
This would save 3 bytes per entry, and in the future it might save even more if we rewrite `igraph_vector_bool_t` as a bit vector.

Reply:
@ntamas `igraph_bool_t` is an `int`, so typically 4 bytes, because of R.

But yes, we might rewrite it in the future to be a `char`.